app/src/ui/history/compare-branch-list-item.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react'
+
+import { Octicon, OcticonSymbol } from '../octicons'
+import { HighlightText } from '../lib/highlight-text'
+import { Branch } from '../../models/branch'
+import { ICompareResult } from '../../lib/git'
+import { Dispatcher } from '../../lib/dispatcher'
+import { Repository } from '../../models/repository'
+
+interface ICompareBranchListItemProps {
+  readonly dispatcher: Dispatcher
+  readonly repository: Repository
+  readonly branch: Branch
+
+  /** Specifies whether this item is currently selected */
+  readonly isCurrentBranch: boolean
+
+  /** The characters in the branch name to highlight */
+  readonly matches: ReadonlyArray<number>
+}
+
+interface ICompareBranchListItemState {
+  readonly compareResult: ICompareResult | null
+}
+
+export class CompareBranchListItem extends React.Component<
+  ICompareBranchListItemProps,
+  ICompareBranchListItemState
+> {
+  public constructor(props: ICompareBranchListItemProps) {
+    super(props)
+
+    this.state = {
+      compareResult: null,
+    }
+  }
+
+  public async componentWillMount() {
+    const compareResult = await this.props.dispatcher.GetCompareResult(
```

Comment:
The performance here is kind of bad. I attempted to use  caching in the `GitStore` but ran into a problem with invalidating entries. If we do use caching, we need to make sure that we update it at the right time to avoid running into issues like never updating the *ahead/behind* info.